universe/worlds/api_worlds.go

```diff
@@ -154,3 +157,91 @@ func (w *Worlds) apiWorldsResolveNameDescription(space universe.Space) (spaceNam
 
 	return name, description, nil
 }
+
+// @Summary Returns spaces based on a searchQuery and categorizes the results
+// @Schemes
+// @Description Returns space information based on a searchquery
+// @Tags spaces
+// @Accept json
+// @Produce json
+// @Param world_id path string true "World ID"
+// @Param query query string true "Space name" example(string)
+// @Success 200 {object} dto.SearchOptions
+// @Success 500 {object} api.HTTPError
+// @Success 400 {object} api.HTTPError
+// @Success 404 {object} api.HTTPError
+// @Router /api/v4/worlds/{world_id}/explore/search [get]
+func (w *Worlds) apiWorldsSearchSpaces(c *gin.Context) {
+	type Query struct {
+		SearchQuery string `form:"query" binding:"required"`
+	}
+
+	inQuery := Query{}
+
+	if err := c.ShouldBindQuery(&inQuery); err != nil {
+		err := errors.WithMessage(err, "Node: apiWorldsSearchSpaces: failed to bind query")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_query", err, w.log)
+		return
+	}
+
+	worldID, err := uuid.Parse(c.Param("worldID"))
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiWorldsSearchSpaces: failed to parse world id")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_world_id", err, w.log)
+		return
+	}
+
+	world, ok := w.GetWorld(worldID)
+	if !ok {
+		err := errors.Errorf("Node: apiWorldsSearchSpaces: space not found: %s", worldID)
+		api.AbortRequest(c, http.StatusNotFound, "world_not_found", err, w.log)
+		return
+	}
+
+	spaces, err := w.apiWorldsFilterSpaces(inQuery.SearchQuery, world)
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiWorldsSearchSpaces: failed to filter spaces")
+		api.AbortRequest(c, http.StatusBadRequest, "failed_to_filter", err, w.log)
+		return
+	}
+
+	c.JSON(http.StatusOK, spaces)
+}
+
+func (w *Worlds) apiWorldsFilterSpaces(searchQuery string, world universe.World) (dto.SearchOptions, error) {
+	predicateFn := func(spaceID uuid.UUID, space universe.Space) bool {
+		name, _, err := w.apiWorldsResolveNameDescription(space)
+		if err != nil {
+			w.log.Error(errors.WithMessagef(err, "failed to resolve name for space: %s", space.GetID()))
```

Comment:
I think it's better return false here, because looks like `Contains()` will return false anyway